tests/drivers/uart/uart_basic_api/src/test_uart_config.c

```diff
@@ -91,14 +91,22 @@ static int test_config_get(void)
 	}
 }
 
+#if CONFIG_SHELL
```

Comment:
i don't see how shell is used here. I don't see any reference to shell apart from enabling it in certain configuration. @dcpleung @nashif do you recall how it is used here? It's been here before new shell was added 4 years ago.

Reply:
My assumption is that the shell is being used to test it interactively (which is kind of odd). We can remove it after this PR.

Reply:
Hi @nordic-krch,  I think like @dcpleung mentioned, we can remove or refine it later.

For now, it works like this in frdm_k64f:
```
Terminal ready
*** Booting Zephyr OS build zephyr-v3.1.0-4262-g9d0e8dbbfb63  ***

------ TESTSUITE SUMMARY START ------

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL


uart:~$ help
Please press the <Tab> button to see all available commands.
You can also use the <Tab> button to prompt or auto-complete all commands or its subcommands.
You can try to call commands with <-h> or <--help> parameter for more information.

Shell supports following meta-keys:
  Ctrl + (a key from: abcdefklnpuw)
  Alt  + (a key from: bf)
Please refer to shell documentation for more details.

Available commands:
  clear                 :Clear screen.
  device                :Device commands
  devmem                :Read/write physical memory"devmem address [width
                         [value]]
  help                  :Prints the help message.
  history               :Command history.
  kernel                :Kernel commands
  resize                :Console gets terminal screen size or assumes default in
                         case the readout fails. It must be executed after each
                         terminal width change to ensure correct text display.
  shell                 :Useful, not Unix-like shell commands.
  test_uart_config_get
  test_uart_configure
  test_uart_fifo_fill
  test_uart_fifo_read
  test_uart_pending
  test_uart_poll_in
  test_uart_poll_out
uart:~$ test_uart_config_get
START - cmd_test_uart_config_get
 PASS - cmd_test_uart_config_get in 0.000 seconds
===================================================================
uart:~$
uart:~$ test_uart_configure
 PASS - cmd_test_uart_configure in 0.000 seconds
===================================================================
uart:~$
uart:~$ test_uart_fifo_read
START - cmd_test_uart_fifo_read
Please send characters to serial console
 PASS - cmd_test_uart_fifo_read in 0.000 seconds
===================================================================
This is a FIFO test.
```